HTTP Server.core/Reply.cs

```diff
@@ -1,10 +1,12 @@
-﻿using System.Text;
+﻿using System;
+using System.Linq;
+using System.Text;
 
 namespace HTTPServer.core
 {
     public static class Reply
     {
-        public static byte[] HandleReply(byte[] request, IDirectoryContents directoryContents)
+        public static byte[] HandleReply(byte[] request, IDirectoryContents directoryContents, IFileContents fileContents)
```

Comment:
i get the segregation here but look at how many places you had to update when you added this.  plus, working with directories and files are conceptually similar. can these 2 things be unified into a larger concept that wouldn't force you to update this method?  really, it seems like this method is asking for too much means that details of its internals are leaking out. anywhere i call it, i now have to provide more arguments.